src/sentry/tasks/merge.py

```diff
@@ -204,12 +204,22 @@ def rehash_group_events(group_id, transaction_id=None, **kwargs):
     delete_group.delay(group.id)
 
 
+def _get_event_environment(event, project):
+    from sentry.models import Environment
+
+    return Environment.get_or_create(
```

Comment:
It might be good to use `get_for_organization_id` here because the environment should already be created (we're not moving it between projects) and `get_or_create` always attempts to create the join table and this is called in a loop.

It might even be worth considering caching this in memory, race conditions be damned

Reply:
> environment should already be created

Famous last words. 😬

Reply:
Yeah 😅